### ISSUE TITLE
Linux pip installs from are downloading OS X binary package

### ISSUE THREAD

**Issue by rhyolight** [numenta/nupic#1734]
### [CentOS 6.5 (AWS AMI)](http://lists.numenta.org/pipermail/nupic-hackers_lists.numenta.org/2015-January/003035.html)

```
[root@ip-10-36-9-98 ~]# lsb_release -a
LSB Version:    :base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch:graphics-4.0-amd64:graphics-4.0-noarch:printing-4.0-amd64:printing-4.0-noarch
Distributor ID:    CentOS
Description:    CentOS release 6.5 (Final)
Release:    6.5
Codename:    Final

[root@ip-10-36-9-98 ~]# pip install nupic
Downloading/unpacking nupic
  Downloading nupic-0.0.33.macosx-10.9-intel.tar.gz (7.1MB): 7.1MB downloaded
  Running setup.py egg_info for package nupic
    Traceback (most recent call last):
      File "<string>", line 16, in <module>
    IOError: [Errno 2] No such file or directory: '/tmp/pip-build-root/nupic/setup.py'
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 16, in <module>

IOError: [Errno 2] No such file or directory: '/tmp/pip-build-root/nupic/setup.py'

----------------------------------------
Command python setup.py egg_info failed with error code 1 in /tmp/pip-build-root/nupic
Storing complete log in /root/.pip/pip.log
```

### [ubuntu 12.04 LTS](http://lists.numenta.org/pipermail/nupic-hackers_lists.numenta.org/2015-January/003037.html)

> 3GB Ram
> AMD Athlon 64 X2 Dual Core Processor 3600+ x 2
> 32-bit OS

```
dhora90@linuxlocal:~$ pip install nupic
Downloading/unpacking nupic
  Downloading nupic-0.0.33.macosx-10.9-intel.tar.gz (7.1Mb): 7.1Mb downloaded
  Running setup.py egg_info for package nupic
    Traceback (most recent call last):
      File "<string>", line 14, in <module>
    IOError: [Errno 2] No such file or directory: '/home/dhora90/build/nupic/setup.py'
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 14, in <module>

IOError: [Errno 2] No such file or directory: '/home/dhora90/build/nupic/setup.py'

----------------------------------------
Command python setup.py egg_info failed with error code 1
Storing complete log in /home/dhora90/.pip/pip.log
```

**Comment by rhyolight** [comment]
Reopening until I confirm...

**Comment by rhyolight** [comment]
Got a confirmation from the ML that someone installing from Ubuntu was getting the sdist.

**Comment by earthgecko** [comment]
Just thought I would update this with some additional info, earlier today I installed via pip fine on Ubuntu 14.04 LTS

```
sudo pip install https://s3-us-west-2.amazonaws.com/artifacts.numenta.org/numenta/nupic/releases/nupic-0.2.6-cp27-none-linux_x86_64.whl
```

```
~$ uname -a
Linux mc11 3.16.0-46-generic #62~14.04.1-Ubuntu SMP Tue Aug 11 16:27:16 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

However with Ubuntu 12.04 LTS the same fails, the all the deps c++, python-devel and python-setuptools are available (both machines have the same puppet build, just different OS versions

```
~# uname -a
Linux mc10 3.5.0-54-generic #81~precise1-Ubuntu SMP Tue Jul 15 04:02:22 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
```

The failure is:

```
~$ sudo pip install https://s3-us-west-2.amazonaws.com/artifacts.numenta.org/numenta/nupic/releases/nupic-0.2.6-cp27-none-linux_x86_64.whl
Downloading/unpacking https://s3-us-west-2.amazonaws.com/artifacts.numenta.org/numenta/nupic/releases/nupic-0.2.6-cp27-none-linux_x86_64.whl
  Downloading nupic-0.2.6-cp27-none-linux_x86_64.whl (22.4Mb): 22.4Mb downloaded
  Running setup.py egg_info for package from https://s3-us-west-2.amazonaws.com/artifacts.numenta.org/numenta/nupic/releases/nupic-0.2.6-cp27-none-linux_x86_64.whl
    Traceback (most recent call last):
      File "<string>", line 14, in <module>
    IOError: [Errno 2] No such file or directory: '/tmp/pip-ql2ql9-build/setup.py'
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 14, in <module>

IOError: [Errno 2] No such file or directory: '/tmp/pip-ql2ql9-build/setup.py'

----------------------------------------
Command python setup.py egg_info failed with error code 1
Storing complete log in /home/earthgecko/.pip/pip.log
```

**Comment by rhyolight** [comment]
You are installing an on version. We just released 0.3.0 a couple days ago. You should be able to install on Linux with these two lines:

pip install https://s3-us-west-2.amazonaws.com/artifacts.numenta.org/numenta/nupic.core/releases/nupic.bindings/nupic.bindings-0.1.5-cp27-none-linux_x86_64.whl

pip install nupic

**Comment by earthgecko** [comment]
Thanks Matthew - although I did not win there were on Ubuntu 12.04 as there were a number of failure vectors, pip version, gcc-4.8 issues around pycapnp.

I bit the bullet and did an upgrade to Ubuntu 14.04.3 and 0.3.0 as above worked, with one caveat, required ```sudo -H ``` on the pip nupic.bindings command on my set up.  So the upgrade resolved all the 12.04 issues, after all if you cannot install a neocortex layer on your OS it it probably outdated :)

Thanks for info.

**Comment by rhyolight** [comment]
Great, although hopefully instead of using `sudo` you can use the `--user` option with pip to install in a non-system location?
